### PR TITLE
Use state-controlled snackbar to allow easy error messages from async…

### PR DESCRIPTION
**Pull request by anuraaga**
… clows.

Migrates from notistack to our own controlled snackbar. To display errors from RPCs, which happen outside the scope of a component, it's simpler to use our normal redux actions for controlling the message instead of notistack. 

@tacigar Note, this introduces a new pattern I hinted at in the other PR of defining everything about the a state (action, reducer) in a component itself instead of globally, and uses the official `redux-toolkit` and `createSlice` for very low boilerplate with type-safety.

I've also brought in `RootState` from your PR, but put it at `/store` instead of `/types` to follow the pattern of semantic separation. Let me know if you're ok with it.

This is to prepare for #3068

### CODE REVIEW

File: zipkin-lens/package.json [hunk 1]
```diff
@@ -20,8 +20,10 @@
     "@lingui/macro": "3.0.0-7",
     "@lingui/react": "3.0.0-7",
     "@material-ui/core": "^4.1.1",
+    "@material-ui/lab": "^4.0.0-alpha.50",
```
Comment (codefromthecrypt) [inline zipkin-lens/package.json:23]: livin on the edge!

File: zipkin-lens/src/components/App/App.jsx [hunk 2]
```diff
@@ -33,41 +32,41 @@ import { i18n } from '../../util/locale';
...
+                  <I18nProvider i18n={i18n}>
+                    <BrowserRouter basename={BASE_PATH}>
+                      <Layout>
+                        <Route exact path="/" component={DiscoverPage} />
+                        {config.dependency.enabled && (
                           <Route
                             exact
-                            path={['/traces/:traceId', '/traceViewer']}
-                            component={TracePage}
+                            path="/dependency"
```
Comment (codefromthecrypt) [inline zipkin-lens/src/components/App/App.jsx:55]: alphabetically ordering the routes, eh? :D

**Review by codefromthecrypt — APPROVED** [review]
I presume the "this prepares for" implies there's no visual impact to this change, hence no screenshot. Thanks for doing the ground work!

File: zipkin-lens/src/components/App/App.jsx (hunk 2 above)
Comment (anuraaga) [reply zipkin-lens/src/components/App/App.jsx:55]: This is GitHub rendering an indentation change poorly, but no change to the routes :)

**Comment by anuraaga** [comment]
@tacigar Let me go ahead and merge this, happy to move things around later as makes sense :)

**Pull request merged**